deploy/infra/aws/terraform/cloud-native/cluster-infra/local.tf

```diff
@@ -54,5 +54,11 @@ locals {
     }
   ]
 
-  cloud_pipeline_db_configuration = var.create_cloud_pipeline_db_configuration && var.deploy_rds ? var.cloud_pipeline_db_configuration : []
-}
+  cloud_pipeline_db_configuration = var.create_cloud_pipeline_db_configuration && var.deploy_rds ? var.cloud_pipeline_db_configuration : {}
+  
+  pipeline_db_pass  = var.cloud_pipeline_db_configuration["pipeline"].password != null ? var.cloud_pipeline_db_configuration["pipeline"].password : nonsensitive(random_password.this["pipeline"].result)
```

Comment:
I would remove `nonsensitive` for these locals and wrap `cp_deploy_script` output with `nonsensitive`